plugins/command-manager/scripts/populate-agents-index.sh

```diff
@@ -0,0 +1,197 @@
+    },
+    "hostname": "host$((RANDOM % 10000))",
+    "id": "hostid$((RANDOM % 10000))",
+    "ip": "$((RANDOM % 256)).$((RANDOM % 256)).$((RANDOM % 256)).$((RANDOM % 256))",
+    "mac": "$(printf '%02x:%02x:%02x:%02x:%02x:%02x' $((RANDOM % 256)) $((RANDOM % 256)) $((RANDOM % 256)) $((RANDOM % 256)) $((RANDOM % 256)) $((RANDOM % 256)))",
+    "name": "hostname$((RANDOM % 10000))",
+    "network": {"egress": {"bytes": $((RANDOM % 1000001)), "packets": $((RANDOM % 1000001))}, "ingress": {"bytes": $((RANDOM % 1000001)), "packets": $((RANDOM % 1000001))}},
+    "os": {"family": "$family", "full": "$family $version", "kernel": "kernel$((RANDOM % 1000))", "name": "$family", "platform": "$(shuf -e linux windows macos -n 1)", "type": "$family", "version": "$version"},
+    "pid_ns_ino": "$((1000000 + RANDOM % 9000000))",
+    "risk": {"calculated_level": "$(shuf -e low medium high -n 1)", "calculated_score": $(echo "scale=2; $RANDOM % 100" | bc), "calculated_score_norm": $(echo "scale=2; $RANDOM % 100 / 100" | bc), "static_level": "$(shuf -e low medium high -n 1)", "static_score": $(echo "scale=2; $RANDOM % 100" | bc), "static_score_norm": $(echo "scale=2; $RANDOM % 100 / 100" | bc)},
+    "uptime": $((RANDOM % 1000001))
+  }
+EOF
+    )
+    echo "$host"
+}
+
+# Function to inject documents to agents index
+function index_documents() {
+    local data=$1
+    url="http://$IP:$PORT/$INDEX_NAME/_doc"
+    response=$(curl -s -o /dev/null -w "%{http_code}" -u $USERNAME:$PASSWORD -H 'Content-Type: application/json' -d "$data" -X POST $url)
+    if [[ $response -ne 201 ]]; then
+        echo "Error: $response"
+    fi
+}
+
+function parse_args() {
+    while getopts ":n:o:h" opt; do
+        case ${opt} in
+        h)
+            echo "Usage: $0 [-n <number>]"
+            echo "Options:"
+            echo "  -n <number>  Number of documents to generate. If not provided, the script will prompt for the number of docs to generate."
+            echo "  -h           Display this help message"
+            echo "Example: $0 -n 100"
+            echo
+            exit 0
+            ;;
+        n)
+            number=$OPTARG
+            ;;
+        o)
+            log_dir=$OPTARG
+            log_file="$log_dir/populate-agents-index.log"
+            ;;
+        \?)
+            echo "Invalid option: $OPTARG" 1>&2
+            exit 1
+            ;;
+        esac
+    done
+}
+
+# Main function
+function populate_index() {
+    if [[ $number -lt 1 ]]; then
+        echo -n "How many docs do you want to generate? "
+        read -r number
+        if ! [[ "$number" =~ ^[0-9]+$ ]]; then
+            echo "Invalid input. Please enter a valid number."
+            return
+        fi
+    fi
+
+    echo "Waiting for the cluster to be up and running..."
+    if ! wait_for_cluster; then
+        echo "Cluster did not start in time. Exiting."
+        exit 1
+    fi
+
+    echo "Generating and indexing $number docs..."
+
+    for i in $(seq 1 "$number"); do
```

Comment:
The use of `seq` is [deprecated](https://www.cyberciti.biz/faq/bash-for-loop/#:~:text=The%20seq%20command%20to%20create%20standard%20bash%20for%20Loop%20(outdated%20method)).

Consider using a more modern approach: https://www.shellcheck.net/wiki/SC2051

Reply:
Fixed in cc6a865